backend/src/controllers/dish.ts

```diff
@@ -418,14 +367,79 @@ export const returnDish = async (req: Request, res: Response) => {
             function: 'returnDish',
         })
 
-        return res.status(200).json({ message: 'dish returned' })
-    } catch (error: any) {
+        return res.status(200).json({ message: "dish returned" })
+    } catch(error: any) {
         Logger.error({
             module: 'dish.controller',
-            function: 'getDish',
+            function: 'returnDish',
             error,
-            message: 'Error when fetching dish',
+            message: 'Error when returning dish',
         })
         return res.status(500).json({ error: 'internal_server_error', message: error.message })
     }
 }
+
+export const updateDishCondition = async (req: Request, res: Response) => {
+    // get qid and condition
+    let qid = req.query['qid']?.toString()
+    if (!qid) {
+        Logger.error({
+            module: 'dish.controller',
+            message: 'No qid provided',
+            statusCode: 400,
+        })
+        return res.status(400).json({ error: 'bad_request' })
```

Comment:
with error can you also leave a message such that people working on the frontend can understand what went wrong. Just like I have left a message with the error.